tests/repositories/permission_grant_test.py

```diff
@@ -38,3 +38,32 @@ def test_permission_grants_for_user(setup):
         ("perm", "one"),
         ("perm", "two"),
     ]
+
+
+def test_permission_grants_for_group(setup):
+    # type: (SetupTest) -> None
+    with setup.transaction():
+        setup.create_group("one-group")
+
+    permission_grant_repository = setup.sql_repository_factory.create_permission_grant_repository()
+    assert permission_grant_repository.permission_grants_for_group("one-group") == []
+
+    # Build a bit of a group hierarchy with some nested inheritance.
+    with setup.transaction():
+        setup.grant_permission_to_group("perm", "one", "one-group")
+        setup.add_group_to_group("one-group", "parent")
+        setup.grant_permission_to_group("other-perm", "arg", "parent")
+        setup.add_group_to_group("parent", "grandparent")
+        setup.grant_permission_to_group("other-perm", "arg", "grandparent")
+        setup.grant_permission_to_group("other-perm", "*", "grandparent")
+        setup.grant_permission_to_group("perm", "two", "two-group")
+        setup.grant_permission_to_group("another-perm", "foo", "np-group")
+
+    # Check the returned permissions.
+    permission_grants = permission_grant_repository.permission_grants_for_group("one-group")
+    assert sorted([(p.permission, p.argument) for p in permission_grants]) == [
+        ("other-perm", "*"),
+        ("other-perm", "arg"),
```

Comment:
Ah, so guessing that the service account permission grant UI would display duplicates like this? Not that it particularly matters

Reply:
Yeah, it currently does. There's an open Jira ticket to fix that; not sure if I'll get to that this week.